### PR TITLE
Using Algorithm as Simulation

### PR DESCRIPTION
**Pull request by Matthiasd23**
# Pull Request Template

## Description
Slight refactor right now

## Jira Ticket
https://repowerednl.atlassian.net/browse/REP-

## Checks
- [x] I have added unit tests
- [x] I have tested this locally
- [x] Does this change introduce a significant decision? If yes, has an ADR been added or updated?

### CODE REVIEW

File: src/marloes/algorithms/ray.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        super().__init__(config, save_energy_flows)  # Initialize the valley/epochs
+        self.config = (
+            SACConfig()
+            .environment(env=lambda _: self.valley)
+            .multi_agent(
+                policies={
+                    agent.id: (
+                        None,
+                        self.valley.observation_space,
+                        self.valley.action_space,
```
Comment (Matthiasd23) [inline src/marloes/algorithms/ray.py:18]: This is allowed, we need to look at what they are, if we should set them ourselves and what the policies are supposed to be. (this was suggested by co-pilot).

**Review by Matthiasd23 — COMMENTED** [review]
I think merge this and work from here? Otherwise this ticket might become pretty big.

File: src/marloes/algorithms/maddpg.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from .base import Algorithm
+
+
+class MADDPG(Algorithm):
```
Comment (LucasVelvis) [inline src/marloes/algorithms/maddpg.py:4]: Are we going to implement this? We will use the library for this right?

File: src/marloes/validation/validate_config.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 def validate_config(config: dict) -> str:
     """Validate the configuration dictionary."""
     # Check if the algorithm is valid
-    valid_algorithms = ["model_based", "model_free", "solver"]
+    valid_algorithms = ["model_based", "model_free", "solver", "maddpg"]
```
Comment (LucasVelvis) [inline src/marloes/validation/validate_config.py:10]: Maybe we should rename solver to priorities?

**Review by LucasVelvis — APPROVED** [review]
Looks good, a few comments

File: src/marloes/algorithms/maddpg.py (hunk 2 above)
Comment (Matthiasd23) [reply src/marloes/algorithms/maddpg.py:4]: Yes, I thought importing it into this class.

File: src/marloes/validation/validate_config.py (hunk 3 above)
Comment (Matthiasd23) [reply src/marloes/validation/validate_config.py]: Done

**Pull request merged**